expression/expression.go

```diff
@@ -1026,6 +1027,12 @@ func scalarExprSupportedByTiKV(sf *ScalarFunction) bool {
 		case tipb.ScalarFuncSig_RandWithSeedFirstGen:
 			return true
 		}
+	case ast.Regexp, ast.RegexpLike:
+		_, collation := sf.Function.CharsetAndCollation()
```

Comment:
Why not check both charset and collation like `func (re *regexpBaseFuncSig) isBinaryCollation() `?